src/time_zone_impl.cc

```diff
@@ -31,9 +31,12 @@ using TimeZoneImplByName =
 TimeZoneImplByName* time_zone_map = nullptr;
 
 // Mutual exclusion for time_zone_map.
-// This mutex is intentionally "leaked" to avoid the static deinitialization
-// order fiasco (std::mutex's destructor is not trivial on many platforms).
-std::mutex* time_zone_mutex = new std::mutex;
+std::mutex& TimeZoneMutex() {
+  // This mutex is intentionally "leaked" to avoid the static deinitialization
+  // order fiasco (std::mutex's destructor is not trivial on many platforms).
+  static std::mutex time_zone_mutex;
```

Comment:
Won't this still have its destructor run?  That is, there is no "leak" here.

Reply:
Of course, my mistake. Returned to making this a pointer (as it was in the original PR) to avoid SDOF.